src/main/scala/akka/persistence/cassandra/query/journal/EventsByPersistenceIdPublisher.scala

```diff
@@ -0,0 +1,91 @@
+package akka.persistence.cassandra.query.journal
+
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+
+import com.datastax.driver.core.Session
+
+import akka.actor.Props
+import akka.persistence.PersistentRepr
+import akka.persistence.cassandra.EventsByPersistenceIdIterator
+import akka.persistence.query.EventEnvelope
+import akka.serialization.SerializationExtension
+
+private[journal] object EventsByPersistenceIdPublisher {
+  def props(
+      persistenceId: String,
+      fromSeqNr: Long,
+      toSeqNr: Long,
+      refreshInterval: Option[FiniteDuration],
+      maxBufSize: Long,
+      session: Session,
+      config: CassandraReadJournalConfig): Props =
+    Props(
+      new EventsByPersistenceIdPublisher(
+        persistenceId,
+        fromSeqNr,
+        toSeqNr,
+        refreshInterval,
+        maxBufSize,
+        session,
+        config))
+}
+
+// TODO: Decouple database.
+// TODO: Query index tables instead.
+// TODO: Generic message iterator to handle different tables etc.
+private[journal] class EventsByPersistenceIdPublisher(
```

Comment:
EventsByPersistenceId query implementation. It uses the original akka.messages table definition. It is relatively thoroughly tested (see tests somewhere below).